### ISSUE TITLE
Synchronous http requests are not supported

### ISSUE THREAD

**Issue by mistikk** [facebook/react-native#27121]
When we try to make Synchronous XMLHttpRequest request, it throws error on release mode. But it is working on debug mode. I couldn't any solution about that. Also i cannot change Synchronous XMLHttpRequest with async/await or promises because this code part is in the another 3rd party library and has a lot of dependency.

```
let request = new XMLHttpRequest();

request.open('POST', url, false);
```

You can check to codes https://github.com/realm/realm-js/blob/38e9ced29ec56d2367b15b0d5974bc7f96953383/lib/browser/rpc.js#L253

![image](https://user-images.githubusercontent.com/8211075/68212062-d7882f00-ffe9-11e9-8ee2-bad132f81581.png)

React Native version: 0.61.2

**Comment by barakmichaely** [comment]
@mistik my guess is that this is most likely related to one of the core concepts in React Native, which is that:

- the javascript layer communicates with the native layer *asynchronously*.  

This means that there’s certain things you just can’t do synchronously. It’s one of the limitations of React Native.

If it was your code, you’d have to have to change it to be asynchronous. The React Native docs show how to use `XmlHttpRequest` asynchronously:
https://facebook.github.io/react-native/docs/network#using-other-networking-libraries

However, quick google search found that Realm is capable of working with react native:
https://realm.io/docs/javascript/latest/

If that doesn’t work for you, then you should open an issue on the Realm Github repo, because the issue is related to their library, not to react native.

**Comment by mistikk** [comment]
i already open issue for realm repo but i am asking another things in here, why sync XmlHttpRequest not working with release version. it was working before 0.60 version, what changed about javascript layer

**Comment by barakmichaely** [comment]
I stand corrected, turns out it was allowed but now it’s deprecated. This change was made to the javascript spec in general, even on the web.

Here’s a quote from the actual spec:

> Synchronous XMLHttpRequest outside of workers is in the process of being removed from the web platform as it has detrimental effects to the end user’s experience. (This is a long process that takes many years.) 

https://xhr.spec.whatwg.org/


The reason behind this change is that because Javascript is single threaded, anytime you make a long synchronous request everything else in your app will freeze until the request is finished. This can be really bad for user experience, and it means you can’t have other tasks or requests running in the background, and your UI won’t be able to update.

Especially with 3rd party libraries, you don't want this side effect because you can’t even control it. 

The quote talks about allowing synchronous requests in web workers, and that is because workers run on different threads. (React Native doesn’t currently have official support for web workers.)

In short: 
Synchronous requests are a bad idea in Javascript (because of single-threadedness) and they aren’t going to come back.
If you depend on a library that uses them, you’ll have to stick with an earlier version of React Native until the library gets updated.
